src/middleware/authValidation.js

```diff
@@ -99,9 +101,26 @@ const resetPassValidator = async (req, res, next) => {
     res.status(400).send(error.message);
   }
 };
+const authorize = (roles) => asyncHandler(async (req, res, next) => {
+  const authHeader = await req.get('Authorization');
+  if (!authHeader) {
+    return res.status(401).json({ error: 'No token provided!' });
+  }
+
+  const token = authHeader.split(' ')[1];
+  const decodedToken = jwt.verify(token, process.env.SECRET_KEY);
+  const role = decodedToken.role;
+  if (!roles.includes(role)) {
+    return res.status(401).json({
+      error: 'Access denied! You are not allowed to perform this operation.'
+    });
+  }
+  next();
+});
```

Comment:
On line 105, change `const authHeader = await req.get('Authorization')` to `const authHeader = await req.get('auth-token')`. This is because swagger doesn't allow to name a header Authorization and remember you need to embed the token in the headers